aws/resource_aws_db_event_subscription.go

```diff
@@ -95,19 +101,25 @@ func resourceAwsDbEventSubscriptionCreate(d *schema.ResourceData, meta interface
 
 	log.Println("[DEBUG] Create RDS Event Subscription:", request)
 
-	_, err := rdsconn.CreateEventSubscription(request)
-	if err != nil {
+	output, err := conn.CreateEventSubscription(request)
+	if err != nil || output.EventSubscription == nil {
 		return fmt.Errorf("Error creating RDS Event Subscription %s: %s", name, err)
 	}
 
+	d.SetId(aws.StringValue(output.EventSubscription.CustSubscriptionId))
```

Comment:
nit: `d.SetId` should handle pointers fine (unlike `d.Set`, IIRC)

Reply:
I'd like to do some out of scope testing, but at first look `d.SetId()` does not have the same nil value reflection as `d.Set()`:

https://github.com/hashicorp/terraform/blob/2dc7241df0245b2be005946b4bc903dfab496c71/helper/schema/resource_data.go#L246-L249

https://github.com/hashicorp/terraform/blob/2dc7241df0245b2be005946b4bc903dfab496c71/helper/schema/resource_data.go#L171-L187

`InstanceState.ID` is `string` so it'd likely cause an interface panic still if I'm understanding the code correctly.

https://github.com/hashicorp/terraform/blob/2dc7241df0245b2be005946b4bc903dfab496c71/terraform/state.go#L1614